source/fundamentals/utf8-validation.txt

```diff
@@ -0,0 +1,116 @@
+.. _nodejs-utf-8-validation:
+
+================
+UTF-8 Validation
+================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to enable or disable the {+driver-short+}'s
+**UTF-8** validation feature. **UTF-8** is a character encoding specification
+that ensures compatibility and consistent presentation across most operating
+systems, applications, and language character sets. By default, the driver
+checks documents for any characters that are not encoded in a valid UTF-8
+format when it transfers data between your application and MongoDB.
+
+If you enable validation, the driver substitutes invalid UTF-8 characters
+with valid UTF-8 ones when transferring data between your application and
+MongoDB. The validation adds processing overhead since it needs to verify the
+data and to make substitutions when necessary.
+
+If you disable validation, your application can avoid the processing
+overhead caused by validation, but cannot guarantee consistent presentation of
+invalid UTF-8 data.
+
+Read the sections below to learn how to set UTF-8 validation using the
+{+driver-short+}.
+
+.. note::
+
+   The {+driver-short+} automatically substitutes invalid UTF-8 characters
+   in your application before sending them to MongoDB regardless of whether
+   you enable validation. You can only retrieve invalid UTF-8 characters
+   from MongoDB.
+
+.. _nodejs-specify-utf-8-validation:
+
+Specify the UTF-8 Validation Setting
+------------------------------------
+
+You can specify whether the driver should perform UTF-8 validation by
+defining the ``enableUtf8Validation`` setting in the options parameter
+when you create a client, reference a database or collection, or call a
+CRUD operation. By default, the {+driver-short+} enables UTF-8 validation.
+
+.. code-block:: javascript
+
+   // disable UTF-8 validation on the client
+   new MongoClient('<connection uri>', { enableUtf8Validation: false });
+
+   // disable UTF-8 validation on the database
+   client.db('<database name>', { enableUtf8Validation: false });
+
+   // disable UTF-8 validation on the collection
+   db.collection('<collection name>', { enableUtf8Validation: false });
+
+   // disable UTF-8 validation on a specific operation call
+   await collection.findOne({ title: 'Cam Jansen'}, { enableUtf8Validation: false });
+
+If your application reads invalid UTF-8 from MongoDB while the
+``enableUtf8Validation`` option is enabled, it throws a ``BSONError`` that
+contains the following message:
```

Comment:
[comment] I thought that it would just make a substitution based on this line: "If you enable validation, the driver substitutes invalid UTF-8 characters with valid UTF-8 ones when transferring data between your application and MongoDB." 

Suggest introducing this behavior in that second paragraph that starts "If you enable validation,..."

Reply:
Oops, I think there's a mistake in the paragraph that starts "if you enable validation" as mentioned in my [other comment](https://github.com/mongodb/docs-node/pull/280/files#r793102498). It doesn't substitute when retrieving data with invalid UTF-8, it fails. If the setting is enabled, the error is thrown which means it doesn't complete the operation. I can include that information about the error if it wasn't clear. Whenever you throw an error in an application, you interrupt whatever operation you were attempting.

I wonder if providing a matrix/table would be helpful or confusing. Something like this:


```
                         |  Read Operations | Write Operations
-----------------------------------------------------
Validation on:  |                               |
-----------------------------------------------------
Validation off:  |                              |
```

Reply:
Got it. The table would definitely be helpful for me!

Reply:
I'm going to hold off on this for now because I think my latest commit is more correct and clear, which would make this redundant.